### PR TITLE
Add customLegendTypes prop to customise covered legend types

### PR DESCRIPTION
**Pull request by Clebal**
Added `customLegendTypes` prop in LegendWidget and LegendWidgetUI to customise legend types.

Example:

```javascript
const CUSTOM_LEGEND_TYPES = {
      'my-custom-legend': MyCustomLegend
}

function App {
      return <LegendWidget customLegendTypes={CUSTOM_LEGEND_TYPES} />
}

function MyCustomLegend () {
      return <p>Test</p>
}

//

dispatch(
  updateLayer({
    id: '...',
    layerAttributes: {
      legend: {
        type: 'my-custom-legend',
        ...
      }
    }
  })
)
```

Result:

image

### CODE REVIEW

**Comment by Clebal** [comment]
After talking with @borja-munoz, we decided to use `customLegendTypes` to allow to only add custom legend types instead of having the possibility to add and modify the current ones. When the legend type isn't one of the covered by default, the LegendWidgetUI checks the `customLegendTypes` object, if it doesn't exists, it renders an error message.

image

**Review by zbigg — COMMENTED** [review]
LGTM, not sure if we need more approvals

**Comment by Clebal** [comment]
@zbigg yes, we need one more approval since this is an enhancement. Btw, you commented, not approved.

**Review by zbigg — APPROVED** [review]

File: CHANGELOG.md
Comment (VictorVelarde) [inline CHANGELOG.md:10]: I think we don't need this entry, as presenting error label is part or the implementation

File: packages/react-ui/src/widgets/legend/LegendWidgetUI.js
Comment (VictorVelarde) [inline packages/react-ui/src/widgets/legend/LegendWidgetUI.js:85]: Maybe use PropTypes.shape with exact props ?

File: packages/react-ui/src/widgets/legend/LegendWidgetUI.js
Comment (VictorVelarde) [inline packages/react-ui/src/widgets/legend/LegendWidgetUI.js:208]: Ok, makes sense to keep existing ones, 👍🏻

**Review by VictorVelarde — APPROVED** [review]
LGTM

File: CHANGELOG.md
Comment (Clebal) [reply CHANGELOG.md:10]: Ok, I'll remove it

**Pull request merged**
